src/core/src/model.cpp

```diff
@@ -443,6 +443,14 @@ void ov::Model::replace_parameter(size_t parameter_index, const shared_ptr<ov::o
     m_parameters[parameter_index] = parameter;
 }
 
+void ov::Model::set_parameter(size_t parameter_index, const shared_ptr<ngraph::op::Parameter>& parameter) {
+    m_parameters[parameter_index] = parameter;
+}
+
+void ov::Model::set_result(size_t index, const shared_ptr<ngraph::op::Result>& result) {
+    m_results[index] = result;
+}
+
```

Comment:
@itikhono, consider adding these methods. There is no good alternative in the current API.